grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/rule/RuleGraph.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql.internal.reasoner.rule;
+
+import ai.grakn.GraknGraph;
+import ai.grakn.concept.Label;
+import ai.grakn.concept.OntologyConcept;
+import ai.grakn.concept.Rule;
+import ai.grakn.concept.Type;
+import ai.grakn.graql.Graql;
+import ai.grakn.graql.internal.reasoner.atom.Atom;
+import ai.grakn.graql.internal.reasoner.query.ReasonerQueryImpl;
+import ai.grakn.util.Schema;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Stack;
+import java.util.stream.Stream;
+
+import static ai.grakn.graql.Graql.var;
+
+/**
+ *
+ * <p>
+ * Convenience class providing methods for operating with the rule graph.
+ * </p>
+ *
+ * @author Kasper Piskorski
+ *
+ */
+public class RuleGraph {
```

Comment:
Would it not be better to have a static `RuleUtil` to house these methods. I ask this because in many places you create a new `RuleGraph` and just throw it away. So the extra object allocation is really not needed.

[*] -- Places where you create a rule graph and then throw it away.